src/Neo/Ledger/TransactionVerificationContext.cs

```diff
@@ -50,15 +51,20 @@ public void AddTransaction(Transaction tx)
         /// Determine whether the specified <see cref="Transaction"/> conflicts with other transactions.
         /// </summary>
         /// <param name="tx">The specified <see cref="Transaction"/>.</param>
+        /// <param name="conflictingTxs">The list of <see cref="Transaction"/> that conflicts with the specified one and are to be removed from the pool.</param>
         /// <param name="snapshot">The snapshot used to verify the <see cref="Transaction"/>.</param>
         /// <returns><see langword="true"/> if the <see cref="Transaction"/> passes the check; otherwise, <see langword="false"/>.</returns>
-        public bool CheckTransaction(Transaction tx, DataCache snapshot)
+        public bool CheckTransaction(Transaction tx, IEnumerable<Transaction> conflictingTxs, DataCache snapshot)
         {
             BigInteger balance = NativeContract.GAS.BalanceOf(snapshot, tx.Sender);
             senderFee.TryGetValue(tx.Sender, out var totalSenderFeeFromPool);
 
-            BigInteger fee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
-            if (balance < fee) return false;
+            BigInteger expectedFee = tx.SystemFee + tx.NetworkFee + totalSenderFeeFromPool;
+            foreach (var conflictTx in conflictingTxs.Where(c => c.Sender.Equals(tx.Sender)))
+            {
+                expectedFee -= (conflictTx.NetworkFee + conflictTx.SystemFee);
+            }
+            if (balance < expectedFee) return false;
```

Comment:
perfect and precise now.

maybe remove the `{ } `

```cs
            foreach (var conflictTx in conflictingTxs.Where(c => c.Sender.Equals(tx.Sender)))
                            expectedFee -= (conflictTx.NetworkFee + conflictTx.SystemFee);
```

Reply:
Fixed.